platform/shared/BUILD

```diff
@@ -20,3 +26,12 @@ filegroup(
     srcs = [".sdk_version"],
     visibility = ["//visibility:public"],
 )
+
+genrule(
+    name = "version_kt_source",
+    outs = ["BuildConstants.kt"],
```

Comment:
this might be for a future pr but just be aware that this file won't be picked up by the gradle system automatically we might need to configure them via gradle sourceSets https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceSet.html
this might be more relevant later when we want to actually expose this value via our API